example/plugins/c-api/txn_data_sink/txn_data_sink.cc

```diff
@@ -28,89 +28,97 @@
   limitations under the License.
  */
 
-#include <stdio.h>
-#include <unistd.h>
 #include <ts/ts.h>
 
-// This gets the PRI*64 types
-#define __STDC_FORMAT_MACROS 1
-#include <inttypes.h>
+#include <string>
+#include <string_view>
 
-#define PLUGIN_NAME "txn_data_sink"
-#define PCP "[" PLUGIN_NAME "] "
-
-// Activate the data sink if this field is present in the request.
-static const char FLAG_MIME_FIELD[] = "TS-Agent";
-static size_t const FLAG_MIME_LEN   = sizeof(FLAG_MIME_FIELD) - 1;
-
-typedef struct {
-  int64_t total;
-} SinkData;
+namespace
+{
+std::string const PLUGIN_NAME = "txn_data_sink";
```

Comment:
Since these are used with a 'c' api it might be best to keep these as constexpr char const *const.  The std::string implies that these values will end up in the heap.